src/modules/directory.rs

```diff
@@ -104,11 +108,48 @@ pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
         },
     );
 
+    if is_readonly_dir(current_dir.to_str()?) {
+        module.create_segment(
+            "read_only_symbol",
+            &config
+                .read_only_symbol
+                .with_style(Some(config.read_only_symbol_style)),
+        );
+    }
+
     module.get_prefix().set_value(config.prefix);
 
     Some(module)
 }
 
+#[cfg(not(target_os = "windows"))]
+fn is_readonly_dir(path: &str) -> bool {
+    let child = Command::new("sh")
+        .arg("-c")
+        .arg(format!("if [ ! -w {} ]; then exit 1; fi", path))
```

Comment:
Calling out to `sh` once per prompt redraw is super expensive, performance-wise.
Why do this manually on windows but shell out on *nix?

Reply:
> Calling out to `sh` once per prompt redraw is super expensive, performance-wise.
> Why do this manually on windows but shell out on *nix?

In my opinion Starship does a lot of really expensive stuff on every redraw (for example, calling go binary or examining git status). An extra `sh` call is not expensive at all in comparison.

But I can rework it for sure if that's the consensus.

Reply:
It is not as expensive as, say, examining git status for a big repo, but it gets called indiscriminately on each directory, that's why I'm more worried about it than otherwise

Reply:
Well, I do agree. I'll fix this, thanks